webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js

```diff
@@ -34,26 +34,28 @@ class AutoComplete extends React.Component {
   componentDidMount() {
     window.addEventListener('keypress', this.windowKeyPressHandler);
     const {
-      id,
       controller,
-      initialError,
-      initialQuery,
+      searchQuery,
+      isDisabled,
+      error,
+      id,
+      url,
       initialUpdate,
     } = this.props;
-    initialUpdate({
-      id,
-      controller,
-      error: initialError,
-      searchQuery: initialQuery,
-    });
+
+    initialUpdate({ searchQuery, controller, id, isDisabled, error, url });
   }
 
   componentDidUpdate(prevProps) {
     this.handleLoading();
-    const { initialQuery } = this.props;
-    if (prevProps.initialQuery !== initialQuery) {
+    const { searchQuery, trigger } = this.props;
+    const { RESET, CONTROLLER_CHANGED } = TRIGGERS;
+    if (trigger === RESET || trigger === CONTROLLER_CHANGED) {
+      this.handleClear();
+    }
+    if (prevProps.searchQuery !== searchQuery) {
       const typeahead = this._typeahead && this._typeahead.current;
-      typeahead && typeahead.setState({ text: initialQuery });
+      typeahead && typeahead.setState({ text: searchQuery });
```

Comment:
Took me a minute to figure out what this is doing.  I think it would be more readable if it were a single-line `if`:
```javascript
if (typeahead) typeahead.setState({ text: searchQuery });
```
Or a regular `if`, if you prefer.

Also, how/why is `setState` being called on something other than `this`?